### PR TITLE
Add tagging for charm publishing

### PR DESCRIPTION
**Pull request by DomFleischmann**
- Automatically tag the latest commit when a charm gets published
- Update README to describe new github-token requirement
- Update to version 0.2.2

### CODE REVIEW

**Comment by ca-scribner** [comment]
Comments on testing, the CI here has not actually tested the action.  This PR either needs to be a from a branch name that is [allowed to publish](https://github.com/canonical/charmhub-upload-action#branch-selection)(branch/charm-tagging), or this PR should add an override to skip the check and we should use that in the CI here.  Right now the "Test charmhub upload" tests show as green, but they're actually just skipped because this branch name is not a publisher.  #19 documents this further and has some suggested fixes.

**Comment by ca-scribner** [comment]
Once tests are re-enabled, we should add tests to assert the tagging added in this test is working correctly.

File: main.js [hunk 1]
```diff
@@ -139,13 +147,46 @@ const yaml = require('js-yaml');
...
+        } else {
+          core.setFailed(
+            'Input "github-token" is missing, and not provided in environment'
+          );
+        }
+      }
+
+      const octokit = github.getOctokit(githubToken);
+      const tagName = `${track}-rev${charmRevision}`;
+      var tagMessage = `${resourceInfo}channel: ${channel}`;
```
Comment (ca-scribner) [inline main.js:172]: I don't believe we can handle `track`/`channel` the same as we can `charmRevision` or resource revisions.  The revisions are immutable facts, whereas the track/channel is a property that may be added or removed.  I think these should be separate tags.  

For `track` and `channel`, these will be added and subtracted as things change in charmhub.  This action needs to handle that as well.  So when this action creates a `channel` tag of `latest/stable`, it also needs to remove the `latest/stable` tag from whatever other commit has it now.  This issue is visible in the dummy repo tags [latest-rev21](https://github.com/DomFleischmann/doms-dummy-charm/releases/tag/latest-rev21) and [latest-rev22](https://github.com/DomFleischmann/doms-dummy-charm/releases/tag/latest-rev22), which both assert they're in `latest/edge`

**Review by ca-scribner — CHANGES_REQUESTED** [review]

File: main.js (hunk 1 above)
Comment (jnsgruk) [reply main.js:172]: I wouldn't bother trying to attach the `latest/stable`, `latest/edge` etc. to the tags. Charmhub clearly shows which revisions are associated with each channel, so I think tagging with just `rev21` is fine. 

The tag message could be amended each time a given revision is posted to a channel?

Comment (jnsgruk) [reply main.js:172]: And by that I mean, when something is released to a channel, append a line like the following to the tag message:

```
Released to `latest/stable` at 14:59:45 UTC on 08 Feb 2022
```

**Pull request merged**